www/js/lib/zimfile.js

```diff
@@ -110,6 +110,42 @@ define(['xzdec_wrapper', 'util', 'utf8', 'q', 'zimDirEntry'], function(xz, util,
         }
     };
 
+    /**
+     * Reads the whole MIME type list and returns it as a populated Map
+     * The mimeTypeMap is extracted once after the user has picked the ZIM file
+     * and is stored as ZIMFile.mimeTypes
+     * 
+     * @returns {Promise} A promise for the MIME Type list as a Map
+     */
+    ZIMFile.prototype._mimeTypeMap = function() {
+        var typeMap = new Map;
+        return this._readSlice(this.mimeListPos, 256).then(function(data) {
+            // DEV: We have read 256 bytes: increase this if you encounter longer MIME type lists
+            // also change "while (pos < 255)" below
+            if (data.subarray) {
+                var i = 1;
+                var pos = -1;
+                var mimeString;
+                while (pos < 255) {
+                    pos++; 
+                    mimeString = utf8.parse(data.subarray(pos), true);
+                    // If the parsed data is an empty string, we have reached the end of the MIME type list, so break 
+                    if (!mimeString) break;
+                    // Store the parsed string in the Map
+                    typeMap.set(i, mimeString);
+                    i++;
+                    while (data[pos]) {
+                        pos++;
+                    }
+                }
+            }
+            return typeMap;
+        }).fail(function(err) {
+            console.errror('Unable to read MIME type list: ' + err);
```

Comment:
There are too many `r` here ;-)
And it's better to put the error as parameter instead of putting it in the string. It allows the browser dev tools to show it as an object.
I suggest `console.error('Unable to read MIME type list', err);`

Reply:
Ah, thanks for catching the typo. Will do.